packages/suite/src/support/messages.ts

```diff
@@ -275,46 +275,126 @@ export default defineMessages({
         id: 'TR_EXCHANGE_TERMS_1',
+        dynamic: true,
     },
     TR_EXCHANGE_TERMS_2: {
         defaultMessage:
-            "You're using this feature to exchange funds that will be sent to an account under your direct personal control.",
+            "You're exchanging cryptocurrency for your own account. You acknowledge that the provider's policies may require identity verification.",
         id: 'TR_EXCHANGE_TERMS_2',
+        dynamic: true,
     },
     TR_EXCHANGE_TERMS_3: {
         defaultMessage:
-            "You're not using this feature for gambling or any other violation of the provider’s terms of service.",
+            'You understand that cryptocurrency transactions are irreversible and may not be refunded. Thus, fraudulent or accidental losses may be unrecoverable.',
         id: 'TR_EXCHANGE_TERMS_3',
+        dynamic: true,
+    },
+    TR_EXCHANGE_VERIFIED_PARTNERS_HEADER: {
+        defaultMessage: 'Verified partners by Invity',
+        id: 'TR_EXCHANGE_VERIFIED_PARTNERS_HEADER',
+        dynamic: true,
     },
     TR_EXCHANGE_TERMS_4: {
         defaultMessage:
-            'You understand that cryptocurrencies are an emerging financial tool and that regulations may be limited in some areas. This may put you at a higher risk of fraud, theft, or market instability.',
+            'You understand that Invity does not provide this service. {provider}’s terms govern the service.',
         id: 'TR_EXCHANGE_TERMS_4',
+        dynamic: true,
+    },
+    TR_EXCHANGE_LEGAL_HEADER: {
+        defaultMessage: 'Legal notice',
+        id: 'TR_EXCHANGE_LEGAL_HEADER',
+        dynamic: true,
     },
     TR_EXCHANGE_TERMS_5: {
         defaultMessage:
-            'You understand that cryptocurrency transactions are irreversible and you won’t be able to receive a refund for your purchase.',
+            "You're not using this feature for gambling, fraud or any other violation of either Invity’s or the provider's Terms of service, or of any applicable regulations.",
         id: 'TR_EXCHANGE_TERMS_5',
+        dynamic: true,
+    },
+    TR_EXCHANGE_TERMS_6: {
+        defaultMessage:
+            'You understand that cryptocurrencies are an emerging financial tool and that regulations may vary in different jurisdictions. This may put you at a higher risk of fraud, theft, or market instability.',
+        id: 'TR_EXCHANGE_TERMS_6',
+        dynamic: true,
+    },
+    TR_EXCHANGE_DEX_FOR_YOUR_SAFETY: {
+        defaultMessage: 'Exchange {fromCrypto} to {toCrypto} with {provider}',
+        id: 'TR_EXCHANGE_FOR_YOUR_SAFETY',
+        dynamic: true,
+    },
+    TR_EXCHANGE_DEX_CONFIRM: {
+        defaultMessage: 'I’m ready to exchange',
+        id: 'TR_EXCHANGE_CONFIRM',
+        dynamic: true,
+    },
+    TR_EXCHANGE_DEX_SECURITY_HEADER: {
+        defaultMessage: 'Security first with your Trezor',
+        id: 'TR_EXCHANGE_SECURITY_HEADER',
+        dynamic: true,
```

Comment:
We plan to change the format of messages.ts so there is no need to have the key and id duplicated. We can add some script to check that. But so far please be super careful and always ensure that the id is the same as key.

Reply:
resolved